src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/service/IntersectionComputer.java

```diff
@@ -56,25 +57,46 @@ public class IntersectionComputer {
      */
     public IntersectionComputer(String targetFlowId, PathId targetForwardPathId, PathId targetReversePathId,
                                 Collection<FlowPath> paths) {
-        List<Edge> targetPath = paths.stream()
-                .flatMap(path -> path.getSegments().stream())
-                .filter(e -> e.getPathId().equals(targetForwardPathId)
-                        || e.getPathId().equals(targetReversePathId))
-                .map(Edge::fromPathSegment)
-                .collect(Collectors.toList());
+        paths.forEach(path -> {
+            if (path.getPathId().equals(targetForwardPathId) || path.getPathId().equals(targetReversePathId)) {
+                handleTargetPath(path);
+            }
+            if (!path.getFlow().getFlowId().equals(targetFlowId)) {
+                handleAnotherPath(path);
+            }
+        });
+    }
 
-        targetPathEdges = new HashSet<>(targetPath);
-        targetPathSwitches = targetPath.stream()
-                .flatMap(e -> Stream.of(e.getSrcSwitch(), e.getDestSwitch()))
-                .collect(Collectors.toSet());
+    private void handleTargetPath(FlowPath path) {
+        if (path.isOneSwitchFlow()) {
+            targetPathSwitches.add(path.getSrcSwitchId());
+        } else {
+            path.getSegments().forEach(segment -> {
+                targetPathEdges.add(Edge.fromPathSegment(segment));
+                targetPathSwitches.add(segment.getSrcSwitchId());
+                targetPathSwitches.add(segment.getDestSwitchId());
+            });
+        }
+    }
 
-        otherEdges = paths.stream()
-                .filter(e -> !e.getFlow().getFlowId().equals(targetFlowId))
-                .flatMap(path -> path.getSegments().stream())
-                .collect(Collectors.groupingBy(
-                        PathSegment::getPathId,
-                        Collectors.mapping(Edge::fromPathSegment, Collectors.toSet())
-                ));
+    private void handleAnotherPath(FlowPath path) {
```

Comment:
Could you explain what are the target and the other switches? I don't get the purpose of this class.
I think the naming could be improved here. `Another` means one more, as we can say in the bar `another beer!` meaning we ask +1 beer. `Other` means different, we can say in the bar `other beer!` meaning we don't like this one, we want some different beer instead of this one.

Reply:
Regarding naming 'Another' means 'an' + 'other' and applies only to one object. For many objects, it will be 'other'.
Regarding purpose - this class calculates overlapping between flows. Target flow is like the "main flow" to which we are comparing other flows.